linkerd/service-profiles/src/lib.rs

```diff
@@ -25,16 +26,20 @@ pub type Receiver = tokio::sync::watch::Receiver<Profile>;
 
 #[derive(Clone, Debug, Default)]
 pub struct Profile {
-    pub name: Option<Name>,
+    pub addr: Option<LogicalAddr>,
     pub http_routes: Vec<(self::http::RequestMatch, self::http::Route)>,
     pub targets: Vec<Target>,
     pub opaque_protocol: bool,
     pub endpoint: Option<(SocketAddr, Metadata)>,
 }
 
 /// A profile lookup target.
-#[derive(Clone, Debug)]
-pub struct LogicalAddr(pub Addr);
+#[derive(Clone, Debug, Hash, Eq, PartialEq)]
+pub struct LookupAddr(pub Addr);
+
+/// A bound logical service address
```

Comment:
nit: should the comment maybe make it clearer that this is part of the profile _returned_ by a lookup, in contrast to `LookupAddr`, which is the input?